ui/lib/config-ui/addon/components/messages/message-expiration-date-form.hbs

```diff
@@ -0,0 +1,54 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+~}}
+
+<div class="has-top-padding-xs has-bottom-padding-s is-narrow is-flex-align-baseline">
+  <RadioButton
+    class="radio"
+    name={{@attr.name}}
+    id="never"
+    value="never"
+    @value="never"
+    @onChange={{fn (mut @message.endTime) ""}}
+    @groupValue={{this.groupValue}}
+  />
+  <label for="never" class="has-left-margin-xs has-text-black is-size-7">
+    <span class="has-left-margin-xs">
+      Never
+    </span>
+    <p class="has-left-margin-xs has-text-grey is-size-8">
+      This message will never expire unless manually deleted by an operator.
+    </p>
+  </label>
+</div>
+
+<div class="has-top-padding-xs has-bottom-padding-s is-narrow is-flex-align-baseline">
+  <RadioButton
+    class="radio"
+    name="{{@attr.name}}"
+    id="specificDate"
+    value="specificDate"
+    @value="specificDate"
+    @onChange={{(fn (mut @message.endTime) this.formDateTime)}}
+    @groupValue={{this.groupValue}}
+  />
+  <label for="specificDate" class="has-left-margin-xs has-text-black is-size-7">
+    <span class="has-left-margin-xs">
+      Specific date
+    </span>
+    <p class="has-left-margin-xs has-text-grey is-size-8">
+      This message will expire at midnight (local timezone) at the specific date.
+    </p>
+    <div class="has-left-margin-xs control">
+      <Input
+        @type="datetime-local"
+        @value={{if this.formDateTime (date-format this.formDateTime this.localDateTimeString) this.formDateTime}}
```

Comment:
I think one of these values is not correct. If `this.formDateTime` is not present, value is `this.formDateTime`? If we intend the fallback to be falsy I would instead use an empty string or something so it looks intentional